backend-utils/src/lib.rs

```diff
@@ -171,17 +171,30 @@ fn split_by_namespace<F: FieldElement>(
 ) -> BTreeMap<String, Vec<StatementIdentifier>> {
     let mut current_namespace = "".to_string();
 
+    // Publics end up in the empty namespace, but we can find the correct namespace by looking
+    // at the referenced polynomial.
+    let public_to_namespace = pil
+        .public_declarations
+        .iter()
+        .map(|(public_name, public)| {
+            let ns = extract_namespace(&public.polynomial.name);
+            (public_name.clone(), ns)
```

Comment:
```suggestion
            (public_name, ns)
```

checked locally, works

Reply:
Oh, didn't see! But looks like we'll revert anyway after #2050 ;)